source/file_open.c

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/file_open.h>
+#include <stdio.h>
+#ifdef _WIN32
+#    include <windows.h>
+#endif
+
+FILE *aws_fopen(const char *file_path, const char *mode) {
+
+#ifdef _WIN32
+    wchar_t w_file_path[1000];
+
+    /* the default encoding is utf-8 or ascii */
+    MultiByteToWideChar(CP_UTF8, 0, file_path, -1, w_file_path, (int)(strlen(file_path) + 1));
+    wchar_t w_mode[10];
+    MultiByteToWideChar(CP_UTF8, 0, mode, -1, w_mode, (int)(strlen(mode) + 1));
+    FILE *file;
+    if (_wfopen_s(&file, w_file_path, w_mode)) {
+        /* errno will be set */
+        return NULL;
```

Comment:
Is all of this really necessary? seems like [fopen_s() supports utf-8](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-s-wfopen-s?view=msvc-160#unicode-support):
> fopen_s supports Unicode file streams. To open a new or existing Unicode file, pass a ccs flag that specifies the desired encoding to fopen_s

Reply:
I think it's saying the content of the file instead of file name? And I just tested it with [fopen_s](https://github.com/awslabs/aws-c-common/pull/794/commits/dfec173a5990016107f87e3d15435be5c4020484), it failed on windows